### PR TITLE
Improve error handling in CLI

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
Signed-off-by: Juan López Fernández <juanlopez@eprosima.com>

### CODE REVIEW

File: src/cpp/user_interface/arguments_configuration.cpp [hunk 1]
```diff
@@ -131,8 +131,9 @@ ProcessReturnCode parse_arguments(
         std::vector<option::Option> buffer(stats.buffer_max);
         option::Parser parse(usage, argc, argv, &options[0], &buffer[0]);
 
-        if (parse.error())
+        if (parse.error() || parse.nonOptionsCount())
```
Comment (jparisu) [inline src/cpp/user_interface/arguments_configuration.cpp:134]: Show different error msg in this case.
Check file https://github.com/eProsima/Fast-DDS/blob/71d986f16f1e6b8a2bb86a85c6c86d1f7f9532d5/tools/fds/server.cpp#L69 to get an idea how to do it.

**Review by jparisu — CHANGES_REQUESTED** [review]
Very good attempt. However, it would be better to fail knowing what was wrong.

File: src/cpp/user_interface/arguments_configuration.cpp (hunk 1 above)
Comment (juanlofer-eprosima) [reply src/cpp/user_interface/arguments_configuration.cpp]: :+1:

**Review by jparisu — DISMISSED** [review]
LGTM

**Comment by jparisu** [comment]
Please Rebase

**Review by jparisu — APPROVED** [review]
LGTM

**Pull request merged**